### PR TITLE
disable all_to_all_fp8 in intranode 

### PR DESCRIPTION
**Pull request by BurkeHulk**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs
- [ ] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/quantization/fp8.py [hunk 1]
```diff
@@ -22,6 +23,17 @@ def wait(self):
...
 
+def process_group_is_intranode(pg):
+    if pg is None:
+        from torch.distributed.distributed_c10d import _get_default_group
+
+        pg = _get_default_group()
+    local_world_size = int(os.environ["LOCAL_WORLD_SIZE"])
+    group_ranks = list(dist.distributed_c10d._pg_group_ranks[pg].values())
+    group_ranks_node_ids = [rank // local_world_size for rank in group_ranks]
+    return min(group_ranks_node_ids) == max(group_ranks_node_ids)
```
Comment (ver217) [inline colossalai/quantization/fp8.py:34]: There is a simpler method that we can compare device count and world size

File: colossalai/quantization/fp8.py
Comment (ver217) [inline colossalai/quantization/fp8.py:1]: And for other communication op, we don't actually use fp8 even when user set `fp8_communication=True` to get a better overall performance.

File: colossalai/quantization/fp8.py (hunk 1 above)
Comment (BurkeHulk) [reply colossalai/quantization/fp8.py:34]: Is it possible that the all-to-all process group only includes a subset of ranks? 
E.g. Given 2 nodes, 8 gpus each node. all-to-all is performed within each node.

Comment (ver217) [reply colossalai/quantization/fp8.py:34]: Yes, you're right. But the env variable "LOCAL_WORLD_SIZE" is not always reliable. E.g. user may launch with slurm/mpi. Maybe we can use device count to get local world size

Comment (BurkeHulk) [reply colossalai/quantization/fp8.py:34]: I see. But I also found that  local_world_size can be smaller than device_count if program is launched with a subset of gpus. E.g. There are 8 gpus per node but `--nproc_per_node=4`. Unless `CUDA_VISIBLE_DEVICES` is set.

Comment (ver217) [reply colossalai/quantization/fp8.py:34]: Oh, if so we may try difference env variables.
- https://docs.hpc.shef.ac.uk/en/latest/referenceinfo/scheduler/SLURM/SLURM-environment-variables.html#gsc.tab=0- 
- https://docs.open-mpi.org/en/v5.0.x/tuning-apps/environment-var.html

Comment (BurkeHulk) [reply colossalai/quantization/fp8.py:34]: Get. Thanks.

**Review by ver217 — APPROVED** [review]

**Pull request merged**